mk/rustllvm.mk

```diff
@@ -43,6 +43,10 @@ $$(RT_OUTPUT_DIR_$(1))/$$(call CFG_STATIC_LIB_NAME_$(1),rustllvm): \
 	@$$(call E, link: $$@)
 	$$(Q)$$(call CFG_CREATE_ARCHIVE_$(1),$$@) $$^
 
+EXTRA_RUSTLLVM_CXXFLAGS_$(1) = $$(shell echo $$(filter $$(LLVM_ALL_COMPONENTS_$(1)),\
+	$$(LLVM_SUBTARGET_COMPONENTS)) | tr 'a-z' 'A-Z' |\
+	sed -e 's/^ //;s/\([^ ]*\)/\-DLLVM_SUBTARGET_\1/g')
```

Comment:
Perhaps this could just be:

```make
EXTRA_RUSTLLVM_CXXFLAGS_$(1) = $$(LLVM_ALL_COMPONENTS_$(1):%=-DLLVM_COMPONENT_%)
```

That way we could just ferry all components down to the C++ code?